src/pages/index.js

```diff
@@ -20,327 +24,246 @@ import UserInfo from "../components/UserInfo.js";
+  "#modal-confirm-delete",
+  config
+);
 
-//***RENDER CARD FUNCTION***
-function renderCard(cardData) {
-  console.log(cardData);
-  const card = new Card(cardData, "#card", handleImageClick);
-  return card.generateCard();
-}
+//***CHANGE PROFILE IMAGE***
+const avatarEditPopup = new PopupWithForm(
+  "#modal-change-profile",
+  handleAvatarFormSubmit,
+  config
+);
+
+//***NEW API***
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "03277265-2976-4ddf-957d-4d187bc8c844",
+    "Content-Type": "application/json",
+  },
+});
+//***USER INFORMATION FROM THE API***
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo({
+      name: res.name,
+      description: res.about,
+    });
+    userInfo.setUserAvatar(res.avatar);
+  })
+  .catch(console.error);
 
-//***NEW SECTION***
 const cardsContainer = new Section(
-  { items: initialCards, renderer: renderCard },
+  { items: [], renderer: renderCard },
   ".cards__list"
 );
+//*** INITIAL CARDS FROM THE API***
+api
+  .getInitialCards()
+  .then((res) => {
+    //***RENDERER METHOD ON CARDSCONTAINER***
 
-//***RENDER CARDSCONTAINER***
-cardsContainer.rendererItems();
+    cardsContainer.rendererItems(res);
+  })
+  .catch(console.error);
+
+//***PREVIEW IMAGE***
+const previewPopup = new PopupWithImage("#image-preview-modal");
 
 //***POPUPWITHFORM***
-const profileEditModal = new PopupWithForm(
+const profileEditPopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileFormSubmit,
   config
 );
+/*/***FUNCTION HANDLING DELETE SUBMIT***
+function handleDeleteFormSubmit(values) {
+  deleteConfirmationPopup.renderSaving(true);
+  api
+    .deleteCard(values)
+    .then((res) => {
+      const deleteCard = renderCard(res);
+      cardsContainer.deleteItem(deleteCard);
+      formValidators.deleteForm.resetForm();
+      deleteConfirmationPopup.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      deleteConfirmationPopup.renderSaving(false);
+    });
+}
+
+*/
+
+//***FUNCTION HANDLING DELETE CLICK***
+function handleDeleteClick(card) {
+  //console.log(card);
+  deleteConfirmationPopup.open();
+  deleteConfirmationPopup.setCallback(() => {
+    deleteConfirmationPopup.renderSaving(true);
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.deleteCard();
+        deleteConfirmationPopup.close();
+      })
+      .catch(console.error)
+      .finally(() => {
+        deleteConfirmationPopup.renderSaving(false);
+      });
+  });
+}
+
+//***FUNCTION HANDLING LIKE CLICK***
+function handleLikeClick(card) {
+  api
+    .likeCard(card.getId(), card.isLiked)
+    .then((res) => card.toggleLikeCard(res.isLiked))
+    .catch(console.error);
+}
+
+//***FUNCTION HANDLING AVATAR EDIT SUBMIT***
+function handleAvatarFormSubmit(values) {
+  avatarEditPopup.renderSaving(true);
+  api
+    .updateAvatar(values)
+    .then((res) => {
+      userInfo.setUserAvatar(res.avatar);
```

Comment:
Please, reset the avatar form and disable the button in `then` after successful avatar changing to let the user change the avatar again without having to clear the inputs manually


```js
      formValidators.changeImageForm.resetForm();
      formValidators.changeImageForm.disableSubmit();

```